Framework/Algorithms/test/PolarizationEfficiencyCorTest.h

```diff
@@ -486,10 +375,53 @@ class PolarizationEfficiencyCorTest : public CxxTest::TestSuite {
     return names;
   }
 
+  std::shared_ptr<PolarizationEfficiencyCor> createAlgorithm(const std::string &efficiencyMethod,
+                                                             const std::string &method = "",
+                                                             const std::string &analysisMethod = "") {
+    auto alg = std::make_shared<PolarizationEfficiencyCor>();
+    alg->setRethrows(true);
+    alg->initialize();
+    alg->setProperty("OutputWorkspace", OUTPUT_GRP_NAME);
+    alg->setProperty("Efficiencies", createEfficiencies(efficiencyMethod));
+    if (!method.empty()) {
+      alg->setProperty("CorrectionMethod", method);
+    }
+    if (!analysisMethod.empty()) {
+      alg->setProperty("PolarizationAnalysis", analysisMethod);
+    }
+    return alg;
+  }
+
+  void checkOutputWorkspaceGroupSize(size_t expectedSize) {
+    WorkspaceGroup_sptr out = AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(OUTPUT_GRP_NAME);
+    TS_ASSERT_EQUALS(out->size(), expectedSize);
+  }
+
+  MatrixWorkspace_sptr createEfficiencyOutputWorkspace(size_t size = 10, double endX = 10) {
+    auto ws1 = createPointWS(size, 0, endX);
```

Comment:
I think we need to pass through the `kind` parameter from `createEfficiencies` and use that to determine what type of workspaces to create. It looks like originally we would call `createPointWs` for `points` and `point-short`, but if `kind == "histo"` then we should be calling `createHistoWS`.